src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreManagerPackage.cs

```diff
@@ -72,6 +78,13 @@ public sealed class RestoreManagerPackage : AsyncPackage
                 _buildEvents = dte.Events.BuildEvents;
                 _buildEvents.OnBuildBegin += BuildEvents_OnBuildBegin;
 
+                var solutionBuildManager = (IVsSolutionBuildManager5)await GetServiceAsync(typeof(SVsSolutionBuildManager));
+                if (solutionBuildManager != null)
+                {
+                    uint updateSolutionEventsCookie4;
```

Comment:
This package will be there in whole VS session, and only be unloaded when you close VS so package Dispose method is never called which is why i didn't store this cookie and called into unadvise... there is no place i can call into unadvise method here.